test/graphql/queries/system_query_test.rb

```diff
@@ -887,32 +887,51 @@ class SystemQueryTest < ActiveSupport::TestCase
 
     should 'properly sort by rules failed' do
       host2 = FactoryBot.create(:host, org_id: @user.account.org_id)
+      host3 = FactoryBot.create(:host, org_id: @user.account.org_id)
+      # For testing sorting of nil failed_rules
+      FactoryBot.create(:host, org_id: @user.account.org_id)
 
-      # An outdated test result made in the past
-      FactoryBot.create(
+      tr = FactoryBot.create(
         :test_result,
         profile: @profile1,
         host: host2,
-        end_time: 1.minute.ago,
-        score: 100
+        score: 50
       )
 
       FactoryBot.create(
         :rule_result,
         host: host2,
         rule: @profile1.rules.first,
         result: 'fail',
-        test_result: FactoryBot.create(
-          :test_result,
-          profile: @profile1,
-          host: host2,
-          score: 50
-        )
+        test_result: tr
+      )
+
+      FactoryBot.create(
+        :rule_result,
+        host: host2,
+        rule: @profile2.rules.first,
+        result: 'fail',
+        test_result: tr
+      )
+
+      tr2 = FactoryBot.create(
+        :test_result,
+        profile: @profile1,
+        host: host3,
+        score: 50
+      )
+
+      FactoryBot.create(
+        :rule_result,
+        host: host3,
+        rule: @profile1.rules.first,
+        result: 'fail',
+        test_result: tr2
       )
 
       query = <<-GRAPHQL
-      query getSystems($policyId: ID, $sortBy: [String!]) {
-          systems(limit: 50, offset: 1, sortBy: $sortBy) {
+      query getSystems($policyId: ID, $filter: String, $sortBy: [String!]) {
+          systems(limit: 50, offset: 1, search: $filter, sortBy: $sortBy) {
```

Comment:
Without searching for a given profile, we cannot really count failed rules properly as rules from other profiles can leak into (that's what happened for you)

Reply:
This makes sense 👌